src/main/java/org/opensearch/neuralsearch/processor/normalization/ZScoreNormalizationTechnique.java

```diff
@@ -52,12 +72,36 @@ public void normalize(NormalizeScoresDTO normalizeScoresDTO) {
             for (int j = 0; j < topDocsPerSubQuery.size(); j++) {
                 TopDocs subQueryTopDoc = topDocsPerSubQuery.get(j);
                 for (ScoreDoc scoreDoc : subQueryTopDoc.scoreDocs) {
-                    scoreDoc.score = normalizeSingleScore(scoreDoc.score, zscores.stdPerSubquery[j], zscores.meanPerSubQuery[j]);
+                    scoreDoc.score = normalizeSingleScore(
+                        scoreDoc.score,
+                        zscores.stdPerSubquery[j],
+                        zscores.meanPerSubQuery[j],
+                        zscores.maxPerSubQuery[j],
+                        zscores.minPerSubQuery[j]
+                    );
                 }
             }
         }
     }
 
+    @Override
+    public void validateCombinationTechnique(ScoreCombinationTechnique combinationTechnique) throws IllegalArgumentException {
+        switch (combinationTechnique.techniqueName()) {
```

Comment:
I would prefer the class level constant Set with supported comb techniques, then you check with Set.contains